presto-bytecode/src/main/java/com/facebook/presto/bytecode/expression/SetFieldBytecodeExpression.java

```diff
@@ -0,0 +1,144 @@
+    {
+        this(instance, requireNonNull(field, "field is null").getDeclaringClass().getType(), field.getName(), value, field.getType());
+        if (instance == null) {
+            checkArgument(field.getAccess().contains(STATIC), "Field is not static: %s", field);
+        }
+        else {
+            checkArgument(!field.getAccess().contains(STATIC), "Field is static: %s", field);
+        }
+    }
+
+    public SetFieldBytecodeExpression(@Nullable BytecodeExpression instance, ParameterizedType declaringClass, String name, BytecodeExpression value)
+    {
+        this(instance, declaringClass, name, value, value.getType());
+    }
+
+    public SetFieldBytecodeExpression(@Nullable BytecodeExpression instance,
+            ParameterizedType declaringClass,
+            String name,
+            BytecodeExpression value,
+            ParameterizedType fieldType)
+    {
+        super(type(void.class));
+        if (instance != null) {
+            checkArgument(!instance.getType().isPrimitive(), "Type %s does not have fields", instance.getType());
+        }
+        this.instance = instance;
+        this.declaringClass = requireNonNull(declaringClass, "declaringClass is null");
+        this.name = requireNonNull(name, "name is null");
+        this.fieldType = requireNonNull(fieldType, "fieldType is null");
+        this.value = requireNonNull(value, "value is null");
+    }
+
+    @Override
+    public BytecodeNode getBytecode(MethodGenerationContext generationContext)
+    {
+        if (instance == null) {
+            return new BytecodeBlock()
+                    .append(value.getBytecode(generationContext))
+                    .putStaticField(declaringClass, name, fieldType);
+        }
+
+        return new BytecodeBlock()
+                .append(instance.getBytecode(generationContext))
+                .append(value.getBytecode(generationContext))
+                .putField(declaringClass, name, fieldType);
+    }
+
+    @Override
+    protected String formatOneLine()
+    {
+        if (instance == null) {
+            return declaringClass.getSimpleName() + "." + name + " = " + value;
+        }
+        else {
+            return instance + "." + name + " = " + value;
+        }
+    }
+
+    @Override
+    public List<BytecodeNode> getChildNodes()
+    {
+        ImmutableList.Builder<BytecodeNode> children = ImmutableList.builder();
+        if (instance != null) {
+            children.add(instance);
+        }
+        children.add(value);
+        return children.build();
+    }
+
+    private static Field getDeclaredField(Class<?> declaringClass, String name)
+    {
+        requireNonNull(declaringClass, "declaringClass is null");
+        requireNonNull(name, "name is null");
+
+        try {
+            return declaringClass.getField(name);
+        }
+        catch (NoSuchFieldException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)